### ISSUE TITLE
White screen when opening /dev directory

### ISSUE THREAD

**Issue by Zotyamester** [HaikuArchives/GenesisCommander#32]
I found out the followings.
* GenesisPanelView.cpp, line 922:
`item = new CustomListItem(name,mPath.String(),FT_FILE,statbuf.st_size, this);`
* GenesisCustomListItem, the above mentioned constructor is in line 28 to line 51

**Comment by Zotyamester** [comment]
Well, if it's okay to not to display device files, then a tiny modification in line 917 would do the job. It would check if the file is in the /dev folder (or its subfolders) and do the followings accordingly.
However, as /dev is a location for device files it's questionable wheter it should be accessible to the user from GenesisCommander, because the main reason why somebody would use this application (in my opinion) is because of its easy to use user interface, it lets you move/copy/delete or rename files and directories easily.
But as I said, **/dev is a place for specialities, not for regular things. It seems unreasonable to let the user in such areas like that.**
Personally I don't even understand why you can access root directory, it's a dangerzone for ordinary people, as far as I know that's why Tracker doesn't allow you to reach it.

**Comment by ocerman** [comment]
@Zotyamester I actually prefer that /dev and other root directories are visible. I was testing some drivers in haiku and having access to /dev helped me a lot. However it could be made as an option.

I think the root cause of this issue is when genesis commander is trying to get icon of `/dev/console`. You can actually recreate it by `cat /dev/console`

**Comment by ocerman** [comment]
I think this is more like a haiku os bug as an attempt of reading any file should not make screen go blank

**Comment by ocerman** [comment]
I have tested that other file managers (BeFar and Krusader) are having same issue.

**Comment by Zotyamester** [comment]
Yep that's the problem (what I said) and I know that Haiku also has a bug when trying to handle dev files as normal files, but it's still a problem.
But still I don't see the point of starting from / instead of /boot (as Tracker does it) or so.

> I was testing some drivers in haiku and having access to /dev helped me a lot.

Usually when you do such a thing, you have a terminal window opened and that's what you use to check /dev and its files and dirs.

**Comment by ocerman** [comment]
> Yep that's the problem (what I said) and I know that Haiku also has a bug when trying to handle dev files as normal files, but it's still a problem.

I agree. I suggest to report this to the haiku os ticket system (if it is not already reported) and let's make a workaround in the meantime.

> But still I don't see the point of starting from / instead of /boot (as Tracker does it) or so.

I think we can make `/boot`, or `~` as a default path. But I don't think restricting user by disallowing some paths like `/` or `/dev` is a good idea, other orthodox file managers also do not restrict user in this way.

> > I was testing some drivers in haiku and having access to /dev helped me a lot.
> 
> Usually when you do such a thing, you have a terminal window opened and that's what you use to check /dev and its files and dirs.

I personally prefer tui/gui for this purpose.

**Comment by pulkomandy** [comment]
On Sun, Dec 22, 2019 at 01:12:08PM -0800, Ondrej Čerman wrote:
> I think this more a haiku bug as an attempt of reading any file should not make screen go blank

The console is a device that points to the screen. If you open it, it
takes over the screen from app_server, that's by design.

It's annoying when you only want to access attributes, maybe we could
implement a specific mode to open for that?

-- 
Adrien.

**Comment by Zotyamester** [comment]
> Maybe we could
implement a specific mode to open for that?

Sorry I did not quite understand that I think.
If you mean that we should modify Haiku then that seems right, but if not then there'll be some problems.
I could only think about some hacking solutions, like checking if the path is under /dev, because as far as I know the API doesn't distinguish between special and regular files.

**Comment by pulkomandy** [comment]
Yes, I meant modifying Haiku. That's the point of having controlon the operating system, right?
